core/src/main/java/org/hibernate/ogm/query/impl/OGMOrderByParser.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.query.impl;
+
+import java.io.StringReader;
+import java.util.List;
+
+import org.hibernate.HibernateException;
+import org.hibernate.dialect.Dialect;
+import org.hibernate.dialect.function.SQLFunctionRegistry;
+import org.hibernate.engine.jdbc.spi.JdbcServices;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.ogm.model.spi.AssociationOrderBy;
+import org.hibernate.sql.ordering.antlr.ColumnMapper;
+import org.hibernate.sql.ordering.antlr.GeneratedOrderByLexer;
+import org.hibernate.sql.ordering.antlr.OrderByFragmentParser;
+import org.hibernate.sql.ordering.antlr.TranslationContext;
+
+import antlr.RecognitionException;
+
+/**
+ * Extracts the list of {@link AssociationOrderBy} from a JPQL order by fragment.
+ * Applying the low level datastore column names.
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class OGMOrderByParser {
+
+	public static List<AssociationOrderBy> parse(String fragment, SessionFactoryImplementor factory, ColumnMapper columnMapper) {
+
+		TranslationContext context = new TranslationContext() {
+			public SessionFactoryImplementor getSessionFactory() {
+				return factory;
+			}
+
+			public Dialect getDialect() {
+				return factory.getServiceRegistry().getService( JdbcServices.class ).getDialect();
+			}
+
+			public SQLFunctionRegistry getSqlFunctionRegistry() {
+				return factory.getSqlFunctionRegistry();
+			}
+
+			public ColumnMapper getColumnMapper() {
+				return columnMapper;
+			}
+		};
+
+		GeneratedOrderByLexer lexer = new GeneratedOrderByLexer( new StringReader( fragment ) );
+
+		// Perform the parsing (and some analysis/resolution).  Another important aspect is the collection
+		// of "column references" which are important later to seek out replacement points in the
+		// translated fragment.
+		OrderByFragmentParser parser = new OrderByFragmentParser( lexer, context );
+		try {
+			parser.orderByFragment();
+		}
+		catch ( HibernateException e ) {
+			throw e;
+		}
+		catch ( Throwable t ) {
+			throw new HibernateException( "Unable to parse order-by fragment", t );
```

Comment:
Shouldn't this be in the log interface?

Reply:
Actually I don't know, because I think that it is not possible to raise the exception on Hibernate OGM side. Since the expression has been already parsed by Hibernate ORM before this code is executed. In other words, it wouldn't be possible create a API level test to verify the case.

Reply:
Not sure if we are talking about the same thing.
I'm just saying to replace this line with
```
throw log.unableToParseOrderBy( e );
```

Reply:
You don't need to add a test case for it's not possible